scijava/scijava-ops-engine/src/main/java/org/scijava/ops/engine/matcher/simplify/SimplifiedOpInfo.java

```diff
@@ -134,89 +135,14 @@ public double priority() {
-	 * simplification of a higher-priority Op wins out over a simplification of a
-	 * lower-priority Op, all else equal.</li>
-	 * <li>a penalty defined as a lossiness heuristic of this simplification. This
-	 * penalty is the sum of:</li>
-	 * <ul>
-	 * <li>the loss undertaken by converting each of the Op's inputs from the ref
-	 * type to the info type</li>
-	 * <li>the loss undertaken by converting each of the Op's outputs from the info
-	 * type to the ref type</li>
-	 * </ul>
-	 * </ul>
-	 */
-	private static double calculatePriority(OpInfo srcInfo, SimplificationMetadata metadata, OpEnvironment env) {
-		// BASE PRIORITY
-		double base = Priority.VERY_LOW;
-
-		// ORIGINAL PRIORITY
-		double originalPriority = srcInfo.priority();
-
-		// PENALTY
-		double penalty = 0;
-
-		Type[] originalInputs = metadata.originalInputs();
-		Type[] opInputs = metadata.focusedInputs();
-		for (int i = 0; i < metadata.numInputs(); i++) {
-			penalty += determineLoss(env, Nil.of(originalInputs[i]), Nil.of(opInputs[i]));
-		}
-
-		// TODO: only calculate the loss once
-		Type opOutput = metadata.focusedOutput();
-		Type originalOutput = metadata.originalOutput();
-		penalty += determineLoss(env, Nil.of(opOutput), Nil.of(originalOutput));
-
-		// PRIORITY = BASE + ORIGINAL - PENALTY
-		return base + originalPriority - penalty;
-	}
-
-	/**
-	 * Calls a {@code lossReporter} Op to determine the <b>worst-case</b>
-	 * loss from a {@code T} to a {@code R}. If no {@code lossReporter} exists for
-	 * such a conversion, we assume infinite loss.
-	 * 
-	 * @param <T> -the generic type we are converting from.
-	 * @param <R> - generic type we are converting to.
-	 * @param from - a {@link Nil} describing the type we are converting from
-	 * @param to - a {@link Nil} describing the type we are converting to
-	 * @return - a {@code double} describing the magnitude of the <worst-case>
-	 *         loss in a conversion from an instance of {@code T} to an instance
-	 *         of {@code R}
-	 */
-	private static <T, R> double determineLoss(OpEnvironment env, Nil<T> from, Nil<R> to) {
-		Type specialType = Types.parameterize(LossReporter.class, new Type[] { from
-			.getType(), to.getType() });
-		@SuppressWarnings("unchecked")
-		Nil<LossReporter<T, R>> specialTypeNil = (Nil<LossReporter<T, R>>) Nil.of(
-			specialType);
-		try {
-			Type nilFromType = Types.parameterize(Nil.class, new Type[] {from.getType()});
-			Type nilToType = Types.parameterize(Nil.class, new Type[] {to.getType()});
-			LossReporter<T, R> op = env.op("lossReporter", specialTypeNil, new Nil[] {
-				Nil.of(nilFromType), Nil.of(nilToType) }, Nil.of(Double.class));
-			return op.apply(from, to);
-		}
-		catch (OpMatchingException e) {
-			return Double.POSITIVE_INFINITY;
-		}
-	}
-
 	@Override
 	public String implementationName() {
-		return srcInfo.implementationName() + " simplified to a " + opType();
+		return info.implementationName() + "_simple";
```

Comment:
Instead of `_`, which is a legal character in Java class names, why not an invalid character like `|`, to ensure that there cannot be a name clash between `Foo_simple` the class and `Foo_simple` the simplified `Foo`?

Reply:
I figured that this would conflict with `SimplifiedOpInfo.signature()`, but I guess it doesn't (which is great!), so I merged it!